src/modules/DecodeQr/Module.js

```diff
@@ -15,19 +15,17 @@ module.exports = function DoNothing(options, UI) {
 
     var step = this;
 
-    getPixels(input.src, function(err, pixels) {
-
-      if (err) throw err;
-
+    function extraManipulation(pixels){
       var w = pixels.shape[0];
       var h = pixels.shape[1];
       var decoded = jsQR(pixels.data, w, h);
-
-
-      // Tell Image Sequencer that this step is complete
-      options.step.qrval = (decoded) ? decoded.data : 'undefined';
-    });
-
+      var qrValue = (decoded) ? decoded.data : 'undefined';
```

Comment:
I had tried this but it gave undefined each time so I left it to be fixed later...

Reply:
I think its about the library jsqr, maybe we should search for another alternative.

Reply:
Are there any good alternatives?

Reply:
@HarshKhandeparkar @root00198 One of the other reason could be that jsqr function might not have executed and while reading the decoded value as it is not yet decoded it will be showing undefined. Some kind of callback might be needed so that we are reading the correct decoded value

Reply:
I dont think jsQr doesn't accept any callbacks... and that it is a synchronous function.

Reply:
Yeah.